phoenix-shared/src/commonMain/kotlin/fr.acinq.phoenix/controllers/config/ChannelsConfigurationController.kt

```diff
@@ -2,79 +2,67 @@ package fr.acinq.phoenix.controllers.config
 
 import fr.acinq.lightning.channel.ChannelStateWithCommitments
 import fr.acinq.lightning.channel.Normal
-import fr.acinq.lightning.serialization.v1.ByteVector32KSerializer
-import fr.acinq.lightning.serialization.v1.Serialization.lightningSerializersModule
+import fr.acinq.lightning.serialization.v3.Serialization.lightningSerializersModule
 import fr.acinq.phoenix.PhoenixBusiness
 import fr.acinq.phoenix.managers.PeerManager
 import fr.acinq.phoenix.controllers.AppController
-import fr.acinq.phoenix.data.Chain
 import fr.acinq.phoenix.utils.localCommitmentSpec
-import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import kotlinx.serialization.builtins.MapSerializer
 import kotlinx.serialization.json.Json
 import org.kodein.log.LoggerFactory
 
 
 class AppChannelsConfigurationController(
     loggerFactory: LoggerFactory,
     private val peerManager: PeerManager,
-    private val chain: Chain
 ) : AppController<ChannelsConfiguration.Model, ChannelsConfiguration.Intent>(
     loggerFactory = loggerFactory,
     firstModel = ChannelsConfiguration.emptyModel
 ) {
     constructor(business: PhoenixBusiness): this(
         loggerFactory = business.loggerFactory,
         peerManager = business.peerManager,
-        chain = business.chain
     )
 
     private val json = Json {
+        ignoreUnknownKeys = true
         prettyPrint = true
         serializersModule = lightningSerializersModule
         allowStructuredMapKeys = true
     }
 
     init {
-        launch {
+        launch(Dispatchers.Default) {
```

Comment:
So at first I noticed there was a significant slowdown on the android app when opening the channels list. I assumed it was that the new memory model + serialization v3 was slower and I tried using a background thread so that the main thread is not to busy. In the end that's not true, the `collect` operation is fast. This slowdown comes from something else (see below) but I thought it was nice to delegate serialization to a background thread anyway, and maybe something we should try to do more often since it's now possible ? Let me know if this is not a good idea.

Regarding the slowdown, maybe it's an unrelated Compose issue on the Android app, or maybe it's the size of the JSON string. On iOS there's no slowdown when fetching the channels data, but my app does struggle a bit when clicking on a channel and showing the JSON.

Reply:
> I thought it was nice to delegate [heavier work] to a background thread, and maybe something we should try to do more often since it's now possible ? Let me know if this is not a good idea.

It sounds like a good idea. I will test it iOS to make sure there's no issues.

Reply:
After testing, there's indeed performance issues on iOS as well.

Switching back to `v1` fixes that. So let's use v1 for now. Serialization in lightning-kmp is going to be reworked very soon, once it's done `v1` will be removed and phoenix will switch to the new api. This will be done in a future PR.